## __RACE BAR PLOT__

__Submitted by-__

1. Ankita Singh (035011)

2. Prabhleen Kaur Pahwa (035039)

In [ ]:
from pydataset import data
import pandas as pd, numpy as np
import scipy.stats as sps
import statsmodels.api as sm, statsmodels.formula.api as smf
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import matplotlib.animation as animation
import bar_chart_race as bcr

In [ ]:
df = pd.read_csv(r"C:\Users\Prabhleen Kaur Pahwa\Documents\1_FORE\Term 3\MLP\Project_035011_035039.csv", encoding='cp1252')
pd.DataFrame(df).head()

(Parent) ASIN (Child) ASIN         Class                         Title  \
0    B07DLYQ5TS   B07DLYQ5TS  Hanging Pots                   Diamond (5)   
1    B07DLYQ5TR   B07DLYQ5TR  Hanging Pots                Orchid (Black)   
2    B07DLWZML2   B07DLWZML2  Hanging Pots             Glory Railing (6)   
3    B083RBW381   B083RBW381  Hanging Pots          Double Hook Railing    
4    B08CY32Q45   B08CY32Q45      Planters  Rectangular Fence Window (4)   

            SKU  Sessions - Total  Sessions – Total – B2B  \
0  OJ-GDWE-6XI9             13927                      74   
1    Pots_00020              8417                      45   
2    Pots_00007              8466                      65   
3    Pots_00054              7844                      37   
4  G5-XUVN-HLWX             10251                     103   

   Session Percentage - Total  Session Percentage – Total – B2B  \
0                      0.1113                            0.0985   
1                      0.0673                            0.0599   
2                      0.0677                            0.0866   
3                      0.0627                            0.0493   
4                      0.0819                            0.1372   

   Page Views - Total  ...  Featured Offer (Buy Box) Percentage  \
0               20449  ...                               0.9929   
1               12554  ...                               0.9932   
2               13220  ...                               0.9906   
3               12220  ...                               0.9802   
4               15666  ...                               0.9951   

   Featured Offer (Buy Box) Percentage – B2B  Units Ordered  \
0                                     0.9905            585   
1                                     0.9865            445   
2                                     0.9120            379   
3                                     0.9800            330   
4                                     1.0000            349   

   Units Ordered - B2B  Unit Session Percentage  \
0                    7                   0.0420   
1                   11                   0.0529   
2                    3                   0.0448   
3                   13                   0.0421   
4                   12                   0.0340   

   Unit Session Percentage - B2B  Ordered Product Sales  \
0                              0                 231803   
1                              0                 307504   
2                              0                 222126   
3                              0                 203495   
4                              0                 278004   

   Ordered Product Sales - B2B  Total Order Items  Total Order Items - B2B  
0                         2777                556                        6  
1                         7633                397                        9  
2                         1797                379                        3  
3                         8912                298                        9  
4                         9540                291                       10  

[5 rows x 23 columns]

# __BRIEF ON ANALYSIS__

1. The data has been picked up from the *Amazon seller account of KHOJI* (a company selling virgin plastic planters). 


2. The data has 91*23 rows and coloumns, respectively and describes the page sessions, page views, orders and the total sales of each 93 unique SKUs of the company over a period of 1 year (Jan 2022- Dec-2022). 


3. __Descriptive variables__: The rows contain 93 unique SKU's of KHOJI, along with 93 unique Parent ASINs and titles. Further, the products are broadly divided into 5 classes:

      a. Hanging Pots

      b. Planters

      c. Decorative (indoor) Planters

      d. Trays

      e. Water cans 



4. __Quantitative variables__: Following are the quantitaive variables which have been further bifurcated into B2B and normal customers 

     a. *Sessions*: The number of unique visitors to your product page

     b. *Page Views*: Total number of times the product page was looked at (this can include multiple/repeat visits from the same people)

     c. *Units orders*: Total number of units sold

     d. *Ordered product sales*: Total number of units sold

     e. *Total Order Items*: Total orders that contained at least one of this SKU

     f. *Buy Box Percentage*: When a visitor landed on the product page, how often were the seller featured in the buy box (ADD TO CART)

     g. *Unit Session Percentage*: Ratio of units ordered to sessions (very similar to a conversion rate)

In [ ]:
df.columns

Index(['(Parent) ASIN', '(Child) ASIN', 'Class', 'Title', 'SKU',
       'Sessions - Total', 'Sessions – Total – B2B',
       'Session Percentage - Total', 'Session Percentage – Total – B2B',
       'Page Views - Total', 'Page Views – Total – B2B',
       'Page Views Percentage - Total', 'Page Views Percentage – Total – B2B',
       'Featured Offer (Buy Box) Percentage',
       'Featured Offer (Buy Box) Percentage – B2B', 'Units Ordered',
       'Units Ordered - B2B', 'Unit Session Percentage',
       'Unit Session Percentage - B2B', 'Ordered Product Sales',
       'Ordered Product Sales - B2B', 'Total Order Items',
       'Total Order Items - B2B'],
      dtype='object')

# __OBJECTIVES__

**1. For Normal customers**


   1a. Highest and Lowest selling product categories.

   1b. Product categories with the Hightest and Lowest session views and pages views.

   1c. Conversion rate of each product class (numbers of page views and sessions that covert into orders). 

   1d. Managerial implications from the above inferences.

**2. For B2B customer**

   2a. Highest and Lowest selling product class.

   2b. Product class with the Hightest and Lowest session views and pages views.
 
   2c. Conversion rate of each product class (numbers of page views and sessions that covert into orders). 

   2d. Managerial implications from the above inferences.

In [ ]:
df1= df.groupby('Class')[['Units Ordered', 'Total Order Items', 'Sessions - Total', 'Page Views - Total']].sum()
df1

Units Ordered  Total Order Items  Sessions - Total  \
Class                                                                 
Decorative Pots            130                126              5011   
Hanging Pots              2478               2290             69337   
Planters                  1287               1136             36854   
Tray                       292                281             11934   
Water Can                   94                 93              1908   

                 Page Views - Total  
Class                                
Decorative Pots                7256  
Hanging Pots                 102491  
Planters                      56228  
Tray                          17594  
Water Can                      2509

In [ ]:
pip install bar_chart_race

Note: you may need to restart the kernel to use updated packages.


In [ ]:
cols1 = ['Class', 'Units Ordered', 'Total Order Items', 'Sessions - Total', 'Page Views - Total' ]
subsetdf = df[cols1]
subsetdf.head()

Class  Units Ordered  Total Order Items  Sessions - Total  \
0  Hanging Pots            585                556             13927   
1  Hanging Pots            445                397              8417   
2  Hanging Pots            379                379              8466   
3  Hanging Pots            330                298              7844   
4      Planters            349                291             10251   

   Page Views - Total  
0               20449  
1               12554  
2               13220  
3               12220  
4               15666

In [ ]:
bcr.bar_chart_race(df = df1, 
                   filename = None, 
                   n_bars = 10, 
                   steps_per_period = 300, 
                   period_length = 500, 
                   figsize = (4, 3), 
                   title ='Class wise yearly orders (Normal)')

C:\Users\Prabhleen Kaur Pahwa\anaconda3\lib\site-packages\bar_chart_race\_make_chart.py:286: UserWarning: FixedFormatter should only be used together with FixedLocator
  ax.set_yticklabels(self.df_values.columns)
C:\Users\Prabhleen Kaur Pahwa\anaconda3\lib\site-packages\bar_chart_race\_make_chart.py:287: UserWarning: FixedFormatter should only be used together with FixedLocator
  ax.set_xticklabels([max_val] * len(ax.get_xticks()))


# OBSERVATIONS

__For Normal Orders__(For 1 year)

**INFERENCES**

1. The top selling products classes were __hanging pots__, with 2478.0 units and __planters__, with 1287.0 units ordered in a year. On the other hand, the bottom selling product classes were __decorative pots__, with 130 units ordered followed by __water cans__ with just 94 units ordered. 


2. The maximum session and page views were recieved by the __hanging pots__ (69337.0, 102491.0) followed by the __planters__  (36854.0, 56228.0), while the least sessions and page views were of __water can__ (1908.0, 2509.0) and __decorative planters__ (5011.0, 7256.0). 


3. Thus, the conversion rates are: 

   3a. Hanging pots: 2.4% (units ordered/page view total) with 3.5% new customers (units ordered/sessions)

   3b. Planters: 2.2% with 3.4% new customers

   3c. Water cans: 3.7% with 4.9% new customers

   3d. Decorative Planters: 1.79% with 2.5% new customers

   3e. Trays: 1.6% with 2.4% new customers



4. __Trays__ recieved a fair amount of page views (17594) and sessions (11934), however its units ordered were only 292. 


**MANAGERIAL IMPLICATIONS**

1. The company should focus on its advertisements of the products in the hanging pots and the planters category because the conversion rates are high, with the new customer visting the sites fairly equal for the two product categories. This also means that the price of the products in these two categories are fair (similar or maybe less than the competitors) and the reviews are also positive. 


2. The conversion rate of water cans are higher, however the products ordered in the year are very less as compared to the other products. The reason behind this is that the products in the category are very less as compared to the others. Thus, we can safely say that the water can product catergory is also doing well in converting the sessions/page views to orders. 


3. Decorative planters and Trays have almost equal covertion rates. However, they are the lowest among all the product categories. The reason behind this could be high prices of the products as compared to the competitors or the products may have negative reviews, which are resulting in the customer not buying the products. The company can thus, either revise their prices or can work on the reviews (remove the negative reviews and try to get to more positive reviews on the pages of these products). 

In [ ]:
df2= df.groupby('Class')[['Sessions – Total – B2B', 'Page Views – Total – B2B',  'Units Ordered - B2B',  'Total Order Items - B2B']].sum()
df2

Sessions – Total – B2B  Page Views – Total – B2B  \
Class                                                               
Decorative Pots                      24                        27   
Hanging Pots                        357                       548   
Planters                            303                       449   
Tray                                 48                        68   
Water Can                            19                        23   

                 Units Ordered - B2B  Total Order Items - B2B  
Class                                                          
Decorative Pots                    3                        3  
Hanging Pots                      57                       38  
Planters                          46                       42  
Tray                               7                        7  
Water Can                          0                        0

In [ ]:
cols2 = ['Class', 'Sessions – Total – B2B', 'Page Views – Total – B2B',  'Units Ordered - B2B',  'Total Order Items - B2B' ]
subsetdf2 = df[cols2]
subsetdf2.head()

Class  Sessions – Total – B2B  Page Views – Total – B2B  \
0  Hanging Pots                      74                       105   
1  Hanging Pots                      45                        74   
2  Hanging Pots                      65                       125   
3  Hanging Pots                      37                        50   
4      Planters                     103                       160   

   Units Ordered - B2B  Total Order Items - B2B  
0                    7                        6  
1                   11                        9  
2                    3                        3  
3                   13                        9  
4                   12                       10

In [ ]:
bcr.bar_chart_race(df = df2, 
                   filename = None, 
                   n_bars = 10, 
                   steps_per_period = 300, 
                   period_length = 500, 
                   figsize = (4, 3), 
                   title ='Class wise yearly orders (B2B)')

C:\Users\Prabhleen Kaur Pahwa\anaconda3\lib\site-packages\bar_chart_race\_make_chart.py:286: UserWarning: FixedFormatter should only be used together with FixedLocator
  ax.set_yticklabels(self.df_values.columns)
C:\Users\Prabhleen Kaur Pahwa\anaconda3\lib\site-packages\bar_chart_race\_make_chart.py:287: UserWarning: FixedFormatter should only be used together with FixedLocator
  ax.set_xticklabels([max_val] * len(ax.get_xticks()))



# OBSERVATIONS
__For B2B Orders__(For 1 year)

**INFERENCES**

1. The top selling products categories were __hanging pots__, with 57 units and __planters__, with 46 units ordered in a year. On the other hand, the bottom selling product classes were __decorative pots__, with 3 units ordered followed by __water cans__ with just 0 units ordered. 


2. The maximum session and page views were recieved by the __hanging pots__ (357.0, 548.0) followed by the __planters__  (303.0, 449.0), while the least sessions and page views were of __water can__ (19.0, 23.0) and __decorative planters__ (24.0, 27.0). 


3. Thus, the conversion rates are: 

   3a. Hanging pots: 10.4% (units ordered/page view total) with 16% new customers (units ordered/sessions)

   3b. Planters: 10.2% with 15.1% new customers

   3c. Water cans: Nill

   3d. Decorative Planters: 11.1% with 12.5% new customers

   3e. Trays: 10.29% with 14.5% new customers



4. __Trays__ recieved a fair amount of page views (48.0) and sessions (68.0), however its units ordered were only 48. 


# MANAGERIAL IMPLICATIONS

1. The offers provided by KHOJI for B2B customers for the products in the hanging pots and the planters category are fairly good as reflected by the high covertion rates and new customer rates. The company should thus continue with their current price and makerting mix for these two catergories. 


2. The convertion rate of Trays and decorative planters are higher, however the products ordered in the year less. Thus company can focus more on these product categories for B2B customers. They can do so by providing better discounts to the customers. 


3. Water cans have had 0 sale for B2B customers, the company can thus stop all its marketing and advertsing spends on the products on this category for B2B because its clearly not proving any benifits and is only resulting in loss of the money. 

# __CONCLUSION__

Overall, in both normal and B2B customers, hanging pots and planters are the best performing product categories for KHOJI. The marketing mix, price mix and the product reviews of these categories are going well. The company should maintain this and should find new ways to promte more sales in these categories. Maybe even increase the products in these catgories. Further, 

1. For the normal customers, KHOJI should focus on the tray catergories to improve sales as it has a high convertion rate.  

2. For B2B customers, KHOJI should drop the spent on advertising on the water can category and focus more on promoting hanging pots, planters, decorative planters and trays. 



-------**The End**--------